Fortran tests now failing for me on my development machine

One of the recent commits broke the fortran build on my machine.

```
Testing PIO's pnetcdf input / output:
   testing PIO_createfile...
 testing  PIO_freedecomp
 after testing  err_msg = no_error                                                                                                                                                                                                                                                        
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            0
 testing  PIO_freedecomp
 after testing  err_msg = no_error                                                                                                                                                                                                                                                        
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            0
 testing  PIO_freedecomp
 after testing  err_msg = no_error                                                                                                                                                                                                                                                        
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            0
      trying to create with noclobber, error expected ... 
success!
   testing PIO_openfile...  4      trying to open nonexistant file error expected ... 
      trying to write to readonly file, error expected ... 
      trying to open non-netcdf file using netcdf, error expected ... 
PIO2 pio_file.c retry NETCDF
success!
   testing PIO_redef...       trying to enter define mode in define mode, error expected ... 
      trying to enter define mode in closed file, error expected ... 
success!
   testing PIO_enddef...       trying to enter define mode in read only file, error expected ... 
      trying to end define mode in data mode, error expected ... 
      trying to end define mode in closed file, error expected ... 
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            2
success!
   testing PIO netCDF-4 functions...  err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            2
 PIO_initdecomp
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            0
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            0
 PIO_initdecomp
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            0
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            0
 PIO_initdecomp
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            0
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            0
 PIO_initdecomp
 PIO_set_chunk_cache
 testing PIO_get_chunk_cache
 PIO_set_chunk_cache
 testing PIO_get_chunk_cache
 PIO_get_chunk_cache returned       140724603453440      140724624268492   0.00000000    
 PIO_openfile piotest_pnetcdf.nc                                                                                                                                                                                                                                              
 PIO_get_chunk_cache returned       140733193388032      140735887892588   0.00000000    
 PIO_openfile piotest_pnetcdf.nc                                                                                                                                                                                                                                              
 PIO_set_chunk_cache
 testing PIO_get_chunk_cache
 PIO_get_chunk_cache returned       140733193388032      140733440151628   0.00000000    
 PIO_openfile piotest_pnetcdf.nc                                                                                                                                                                                                                                              
 PIO_set_chunk_cache
 testing PIO_get_chunk_cache
 PIO_get_chunk_cache returned       140724603453440      140725828327948   0.00000000    
 PIO_openfile piotest_pnetcdf.nc                                                                                                                                                                                                                                              
 PIO_redef
 PIO_redef
 PIO_redef
 PIO_def_dim
 PIO_redef
 PIO_def_dim
 PIO_def_dim
 PIO_def_var
 PIO_def_dim
 PIO_def_var
 PIO_def_var
 PIO_def_var
 PIO_set_var_chunk_cache
 PIO_set_var_chunk_cache
 PIO_get_var_chunk_cache
 PIO_get_var_chunk_cache ret_val=        -111
 testing PIO_def_var_deflate
 testing PIO_put_att
 PIO_set_var_chunk_cache
 PIO_get_var_chunk_cache
 PIO_get_var_chunk_cache ret_val=        -111
 testing PIO_def_var_deflate
 testing PIO_put_att
 PIO_set_var_chunk_cache
 PIO_get_var_chunk_cache
 PIO_get_var_chunk_cache ret_val=        -111
 testing PIO_def_var_deflate
 testing PIO_put_att
 PIO_get_var_chunk_cache
 PIO_get_var_chunk_cache ret_val=        -111
 testing PIO_def_var_deflate
 testing PIO_put_att
 testing PIO_put_att
 testing PIO_put_att
 testing PIO_put_att
 testing PIO_put_att
 testing PIO_inq_var_deflate
 testing PIO_def_var_deflate
 testing PIO_enddef
 testing PIO_inq_var_deflate
 testing PIO_def_var_deflate
 testing PIO_enddef
 testing PIO_inq_var_deflate
 testing PIO_def_var_deflate
 testing PIO_enddef
 testing PIO_inq_var_deflate
 testing PIO_def_var_deflate
 testing PIO_enddef
 testing PIO_inq_var_deflate
 testing PIO_write_darray
 testing PIO_inq_var_deflate
 testing PIO_write_darray
 testing PIO_inq_var_deflate
 testing PIO_write_darray
 testing PIO_inq_var_deflate
 testing PIO_write_darray
 testing  PIO_closefile
 testing  PIO_closefile
 testing  PIO_closefile
 testing  PIO_closefile
 testing  PIO_freedecomp
 testing  PIO_freedecomp
 after testing  err_msg = no_error                                                                                                                                                                                                                                                        
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            0
 testing  PIO_freedecomp
 after testing  err_msg = no_error                                                                                                                                                                                                                                                        
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            0
 testing  PIO_freedecomp
 after testing  err_msg = no_error                                                                                                                                                                                                                                                        
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            0
 after testing  err_msg = no_error                                                                                                                                                                                                                                                        
 err_msg =no_error                                                                                                                                                                                                                                                         fail_cnt =            2
success!
 
Total failure count: 2
FAILED unit testing.
STOP 1
STOP 0
STOP 0
STOP 0
FAIL: run_tests.sh
==================
1 of 1 test failed
==================
make[3]: *** [Makefile:553: check-TESTS] Error 1
make[3]: Leaving directory '/home/ed/ParallelIO/tests/unit'
```

--- a/tests/unit/ncdf_tests.F90
+++ b/tests/unit/ncdf_tests.F90
@@ -106,18 +106,6 @@ Contains
        return
     end if
 
-    ! Try to enter define mode again
-    if(master_task) write(*,"(6x,A,1x)") "trying to enter define mode in define mode, error expected ... "
-    call mpi_barrier(MPI_COMM_WORLD,ret_val)
-
-    ret_val = PIO_redef(pio_file)
-    if (ret_val .eq. PIO_NOERR) then
-       ! Error in PIO_redef
-       err_msg = "Entered define mode from define mode"
-       call PIO_closefile(pio_file)
-       return
-    end if
-
     ! Leave define mode
     ret_val = PIO_enddef(pio_file)
     if (ret_val .ne. PIO_NOERR) then
@@ -313,7 +301,7 @@ Contains
     print*, 'PIO_set_chunk_cache'
     ret_val = PIO_set_chunk_cache(pio_iosystem%iosysid, iotype, chunk_cache_size, &
          chunk_cache_nelems, chunk_cache_preemption)
-    
+
     ! Should not have worked except for netCDF-4/HDF5 iotypes.
     if (iotype .eq. PIO_iotype_netcdf4c .and. ret_val .ne. PIO_NOERR) then
        err_msg = "Could not set chunk cache"
@@ -339,7 +327,7 @@ Contains
          chunk_cache_nelems_in, chunk_cache_preemption_in)
     print*, 'PIO_get_chunk_cache returned ', chunk_cache_size_in, &
          chunk_cache_nelems_in, chunk_cache_preemption_in
-    
+
     ! Should not have worked except for netCDF-4/HDF5 iotypes.
     if (iotype .eq. PIO_iotype_netcdf4c .or. iotype .eq. PIO_iotype_netcdf4p) then
        if (ret_val .ne. PIO_NOERR) then
@@ -407,7 +395,7 @@ Contains
     print*, 'PIO_set_var_chunk_cache'
     ret_val = PIO_set_var_chunk_cache(pio_file, pio_var, chunk_cache_size, chunk_cache_nelems, &
          chunk_cache_preemption)
-    
+
     ! Should not have worked except for netCDF-4/HDF5 iotypes.
     if (iotype .eq. PIO_iotype_netcdf4c .and. ret_val .ne. PIO_NOERR) then
        err_msg = "Could not set variable chunk cache"
@@ -432,7 +420,7 @@ Contains
     ret_val = PIO_get_var_chunk_cache(pio_file, pio_var, chunk_cache_size_in, &
          chunk_cache_nelems_in, chunk_cache_preemption_in)
     print*, 'PIO_get_var_chunk_cache ret_val=', ret_val
-    
+
     ! Should not have worked except for netCDF-4/HDF5 iotypes.
     if (iotype .eq. PIO_iotype_netcdf4c .or. iotype .eq. PIO_iotype_netcdf4p) then
        if (ret_val .ne. PIO_NOERR) then
@@ -458,7 +446,7 @@ Contains
     end if
 
     ! Try to turn on compression for this variable.
-    print*, 'testing PIO_def_var_deflate' 
+    print*, 'testing PIO_def_var_deflate'
     shuffle = 0
     deflate = 1
 
@@ -627,10 +615,10 @@ Contains
     call PIO_closefile(pio_file)
 
     ! Free decomp
-    print*, 'testing  PIO_freedecomp'    
+    print*, 'testing  PIO_freedecomp'
     call PIO_freedecomp(pio_iosystem, iodesc_nCells)
     call mpi_barrier(MPI_COMM_WORLD,ret_val)
-    
+
     print*, 'after testing  err_msg = '    , err_msg
   End Subroutine test_nc4
 end module ncdf_tests

